<robot name="simple_robot"> <!-- robot name- you decide -->
<!-- sequence : link-joint-link-joint-...-link... -->
<!-- units: meters for distance,rad in case of angles -->
<!-- first link called base_link -->
<link name= "base_link">
	<visual>
		<geometry>
			<cylinder length="0.01" radius="0.2"/>
		</geometry>
		<origin rpy="0 0 0" xyz="0 0 0"/> <!-- rpy: roll pitch yaw -> orientation xyz -> position --> 
		<material name= "base_color">
			<color rgba="1 1 0 1"/> 
		</material>
	</visual>

</link>


<!-- first joint-->
<joint name="joint1" type= "revolute"> <!-- types: revolute and prismatic -->
	<parent link="base_link"/>
	<child link="link1"/>
	<origin xyz ="0 0 0"/>
	<axis xyz="0 0 1"/>
	<limit effort="300" velocity="0.1" lower="-3.14159" upper="3.14159"/>
	<dynamics damping="50" friction="1"/>

</joint>


<!-- link1-->
<link name= "link1">
	<visual>
		<geometry>
			<cylinder length="0.4" radius="0.04"/>
		</geometry>
		<origin rpy="0 0 0" xyz="0 0 0.2"/> <!-- rpy: roll pitch yaw -> orientation xyz -> position --> 
		<material name= "link1_color">
			<color rgba="1 0 0 1"/> 
		</material>
	</visual>

</link>


<!-- second joint-->
<joint name="joint2" type= "revolute"> <!-- types: revolute and prismatic -->
	<parent link="link1"/>
	<child link="link2"/>
	<origin xyz ="0 0 0.4"/>
	<axis xyz="0 1 0"/>
	<limit effort="300" velocity="0.1" lower="-3.14159" upper="3.14159"/>
	<dynamics damping="50" friction="1"/>

</joint>


<!-- link2-->
<link name= "link2">
	<visual>
		<geometry>
			<cylinder length="0.4" radius="0.04"/>
		</geometry>
		<origin rpy="0 1.5707 0" xyz="0.2 0 0"/> <!-- rpy: roll pitch yaw -> orientation xyz -> position --> 
		<material name= "link2_color">
			<color rgba="0 1 0 1"/> 
		</material>
	</visual>

</link>


<!-- third joint-->
<joint name="joint3" type= "revolute"> <!-- types: revolute and prismatic -->
	<parent link="link2"/>
	<child link="link3"/>
	<origin xyz ="0.4 0 0"/>
	<axis xyz="0 1 0"/>
	<limit effort="300" velocity="0.1" lower="-3.14159" upper="3.14159"/>
	<dynamics damping="50" friction="1"/>

</joint>


<!-- link3-->
<link name= "link3">
	<visual>
		<geometry>
			<cylinder length="0.3" radius="0.04"/>
		</geometry>
		<origin rpy="0 1.5707 0" xyz="0.15 0 0"/> <!-- rpy: roll pitch yaw -> orientation xyz -> position --> 
		<material name= "link3_color">
			<color rgba="0 1 0 1"/> 
		</material>
	</visual>

</link>


<!-- fourth joint-->
<joint name="joint4" type= "revolute"> <!-- types: revolute and prismatic -->
	<parent link="link3"/>
	<child link="link4"/>
	<origin xyz ="0.3 0 0"/>
	<axis xyz="0 -1 0"/>
	<limit effort="300" velocity="0.1" lower="-3.14159" upper="3.14159"/>
	<dynamics damping="50" friction="1"/>

</joint>


<!-- link4-->
<link name= "link4">
	<visual>
		<geometry>
			<cylinder length="0.22" radius="0.04"/>
		</geometry>
		<origin rpy="1.5707 0 0" xyz="0.0 0.11 0"/> <!-- rpy: roll pitch yaw -> orientation xyz -> position --> 
		<material name= "link4_color">
			<color rgba="0 1 1 1"/> 
		</material>
	</visual>

</link>


<!-- fifth joint-->
<joint name="joint5" type= "fixed"> <!-- types: revolute and prismatic -->
	<parent link="link4"/>
	<child link="link5"/>
	<origin xyz ="0 0.22 0"/>
	<axis xyz="0 -1 0"/>
	<limit effort="300" velocity="0.1" lower="-3.14159" upper="3.14159"/>
	<dynamics damping="50" friction="1"/>
</joint>
<!-- link5-->
<link name= "link5">
	<visual>
		<geometry>
			<cylinder length="0.0" radius="0.0"/>
		</geometry>
		<origin rpy="0 0 0" xyz="0.0 0 0"/> <!-- rpy: roll pitch yaw -> orientation xyz -> position --> 
		<material name= "link5_color">
			<color rgba="0 1 0 1"/> 
		</material>
	</visual>

</link>




</robot>


